plugins/modules/iso.py

```diff
@@ -256,8 +256,8 @@ def main():
     )
 
     module.deprecate(
-        "The 'results' return value is being renamed to 'record' and changed from list to dict."
-        "Please use 'record' since 'results' will be removed in future release."
+        "The 'results' return value is being renamed to 'record' and changed from list to dict. "
+        "Please use 'record' since 'results' will be removed in future release. "
         "But for now both values are being returned to allow users to migrate their automation.",
```

Comment:
Should we also have "affected module: iso" in the deprecation message (similar to vm)? The same goes for vm_nic

Reply:
Could be a bit confusing, for `iso` and `vm_nic` we are able to implement new way before old way of working is removed. What if I instead replace in vm/snapshot_schedule module (and add to iso/vm_nic) :

```
# remove
# DOCUMENTATION, notes section
    Affected modules are M(scale_computing.hypercore.vm) and M(scale_computing.hypercore.snapshot_schedule).
#    module.deprecate later
        "Affected modules are vm and snapshot_schedule.",

# add
# DOCUMENTATION, notes section
        "For complete list of deprecation changes, please see https://justinc1.github.io/HyperCoreAnsibleCollection/extra/deprecation.html",
#    module.deprecate later
        "For complete list of deprecation changes, please see https://justinc1.github.io/HyperCoreAnsibleCollection/extra/deprecation.html",
```

We add one `deprecation.rst` file, and there we can include details, also an example how new playbook using `vm` module will look like.

Reply:
CC @jtaubenheim - I expect you also have opinion about this.

Reply:
@justinc1 as I said in the slack convo I do not believe we should remove the old instructions on the module pages until 3.0.0 release (which you have noted). My only question is if there will be some note or warning on the module documentation pages (i.e., https://scalecomputing.github.io/HyperCoreAnsibleCollection-docs/modules/snapshot_schedule.html) <--- this is still the old version on AnsibleHyperCoreCollection-docs. But at note that directs users from the affected modules details page to the deprecation page? I don't see any updates to the files `/snapshot_schedule.rst.txt` or `/vm.rst.txt`. My only concern here is that we cannot expect users to regularly check the deprecation page and it needs to be noted on the affected module pages that they need to visit the deprecation page to see the details.

Reply:
@jtaubenheim I didn't correctly understand that you are asking to include link to deprecation page on affected module pages.

It actually is already there, in "notest" section of each module page. I will just paste all links (for my fork):
- https://justinc1.github.io/HyperCoreAnsibleCollection/collections/scale_computing/hypercore/iso_module.html#notes
- https://justinc1.github.io/HyperCoreAnsibleCollection/collections/scale_computing/hypercore/vm_nic_module.html#notes
- https://justinc1.github.io/HyperCoreAnsibleCollection/collections/scale_computing/hypercore/vm_module.html#notes
- https://justinc1.github.io/HyperCoreAnsibleCollection/collections/scale_computing/hypercore/snapshot_schedule_module.html#notes

Common text to all is "[List of deprecation changes](https://justinc1.github.io/HyperCoreAnsibleCollection/extra/deprecation.html#scale-computing-hypercore-deprecation) includes examples to help with transition." Will this be sufficient?

Reply:
Yep that looks good. I just didn't want to rely on people navigating to the deprecation page to know that there are planned changes to the specific modules and having it on the module pages is really just a CYA precaution. Looks good to me!

Reply:
Thank you, merging now.